File: lc_java/lcJava/prefix/CheckSubarraySum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int m = nums.length;
        Map<Integer, Integer> modMap = new HashMap<>();
        modMap.put(0, -1);
        int prefixS = 0;
        
        for (int i = 0; i < m; i++) {
            prefixS += nums[i];
            
            int mod = prefixS % k;
            
            if (modMap.containsKey(mod)) {
                if (i - modMap.get(mod) >= 2) {
                    return true;
                }
            } else {
                modMap.put(mod, i);
            }
        }
        return false;
    }
}

File: lc_js/lcJs/prefix/checkSubarraySum.js
function checkSubarraySum(nums, k) {
    const m = nums.length;
    const modMap = new Map();
    modMap.set(0, -1);
    let prefixS = 0;
    
    for (let i = 0; i < m; i++) {
        prefixS += nums[i];
        
        const mod = prefixS % k;
        
        if (modMap.has(mod)) {
            if (i - modMap.get(mod) >= 2) {
                return true;
            }
        } else {
            modMap.set(mod, i);
        }
    }
    return false;
}